src/extension/sdk/dev_tools/manager.ts

```diff
@@ -420,11 +415,9 @@ class DevToolsService extends StdIOService<UnknownNotification> {
 
 		this.spawnedArgs = this.getDevToolsArgs();
 
-		const { binPath, binArgs } = usingCustomScript(
-			path.join(workspaceContext.sdks.dart, pubPath),
-			this.spawnedArgs,
-			workspaceContext.config?.devtoolsRunScript,
-		);
+		// TODO(helin24): Use daemon instead to start DevTools if internal workspace
```

Comment:
Curious what "daemon" is in this context (or more specifically what its lifetime is and whether it allows us to still use the devtools_server?).

Reply:
This is related to the above comment - externally DevTools can be started by first running `flutter daemon` and then passing a message like `[{"message":"devtools.launch"}]`. Then the daemon will print out the address of the DevTools server and this can be used the same as when DevTools is launched with `pub global`. (Above I mentioned `daemonScript` which does the same thing as `flutter daemon` internally). I may need to know more about how `devtools_server` is used to determine whether everything would be the same otherwise.

Reply:
Ah got it - I wasn't sure if the daemon you meant was similar to the `flutter run` daemon, which would be more tricky as it has a different lifetime (it's once per debug session, whereas the DevTools server and `flutter daemon` can live for the whole editor session).

Right now devtools_server is used for some additional functionality like re-connecting DevTools instances when you start a new debug session. We can't currently run DevTools without the server, although it's possible a lot of its functionality could be migrated away or dropped. For example, embedded views we can refresh ourselves to reconnect them to new apps - this isn't the case for external browser windows we spawn though.

If using the Pub version for now works, sounds like this doesn't need resolving immediately though.